### ISSUE TITLE
Add 'Error' prefix to TF_LITE_REPORT_ERROR

### ISSUE THREAD

**Issue by freddan80** [tensorflow/tensorflow#42626]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): N/A
- Tensorflow version (commit SHA if source): 9578a394a0ebfa5f77f3e3b87f7b7fa266c97103
- Target platform (e.g. Arm Mbed OS, Arduino Nano 33 etc.): Any

**Describe the problem**
It would be great if macro TF_LITE_REPORT_ERROR() prefixed the output with 'Error' or similar. That makes it easier for CI systems to parse for new issues. In a similar fashion, a macro TF_LITE_REPORT_WARNING() would be nice for warning. This could be a good location for the prefix:

```
#define TF_LITE_REPORT_ERROR(reporter, ...)                             \
  do {                                                                  \
    static_cast<tflite::ErrorReporter*>(reporter)->Report(__VA_ARGS__); \
  } while (false)
#else  // TF_LITE_STRIP_ERROR_STRINGS
#define TF_LITE_REPORT_ERROR(reporter, ...)
#endif  // TF_LITE_STRIP_ERROR_STRINGS
```
Let me know if there's a better way to achieve this.

**Please provide the exact sequence of commands/steps when you ran into the problem**
Any error print.

**Comment by advaitjain** [comment]
The potentially biggest issue here would be that we'd like to make sure any change to the logging doesn't adversely affect Mobile.

Can you give a few specific examples of the log messages that you would like to filter and why?

Are the errors that you are trying to filter for in the Micro-only code or in the parts that are shared between Micro and Mobile.

There are a few things to consider:

 * We know that for Micro, the implementation of the testing macros can obfuscate what the error was and depending on what issue you are facing, the solution might be to fix that issue.

 * I can imagine adding a log level to the TF_LITE_REPORT_ERROR macro that could be used for Micro to add a suitable prefix, but leaves the behavior in the shared lite code unchanged, if that is what we need.

 * Alternately, for Micro, we are considering moving to a single global error reporter which may also mean a difference between the error reporting between Mobile and Micro -- another opportunity to roll-in some feature requests.

**Comment by freddan80** [comment]
The main ask is for a 'Error' to appear in the logs when TF_LITE_REPORT_ERROR() is called. One example would be this error:
https://github.com/tensorflow/tensorflow/blob/ff4af376571342beaef64ef4e34b632e0c98e8d8/tensorflow/lite/micro/micro_allocator.cc#L619
By prefixing the resulting printout with 'Error: ' it would be easier for CI systems to pinpoint the issue. In our CI system, and others as well, there are many SW components are involved. All of them could fail. Typically our CI parse on the string 'error' to narrow component and log line that cause the issue, which save manual log analysis time.

**Comment by mansnils** [comment]
Is it so that we are moving away from TF_LITE_REPORT_ERROR() in TFLM and instead use GetMicroErrorReporter() and MicroPrintf() more and more? It seems so far GetMicroErrorReporter() is mostly used in kernel tests.
Assuming it would be used for all logging in TFLM, would it make sense to add a log level for it?
So that specific error prints could have an error prefix.

**Comment by advaitjain** [comment]
Yes, we do want to use MicroPrintf and GetMicroErrorReporter for all logging in TFLM. We can certainly consider adding a log level there, though my hope is that before we go down the path of implementing our own embedded logging library, we also take a look at other options (such as https://pigweed.dev/pw_log/).

Such an investigation is not something that we have currently planned so I am not expecting any big changes to how we do our logging in the near future.

**Comment by jdduke** [comment]
@advaitjain should we move this issue to the TFLM repo?
